src/components/views/elements/AppTile.js

```diff
@@ -98,23 +103,35 @@ export default React.createClass({
         return false;
     },
 
-    componentWillMount: function() {
+    componentWillMount() {
+        window.addEventListener('message', this._onMessage, false);
+        this.updateWidgetContent();
+    },
+
+    // Update widget content
+    updateWidgetContent() {
+        this.setState(this._getInitialState());
+        // Set scalar token on the wUrl, if needed
+        this.setScalarToken();
+    },
+
+    // Adds a scalar token to the widget URL, if required
+    setScalarToken() {
         if (!this.isScalarUrl()) {
             return;
         }
         // Fetch the token before loading the iframe as we need to mangle the URL
-        this.setState({
-            loading: true,
```

Comment:
If this is no longer needed, the partner `loading: false`s should be removed too. But it would be strange to not display this loading state, no? Unless this happens too quickly for a spinner to be worth it?

Reply:
The loading state is now set to true in the _getInitialState function on line 64/65. The loading state spinner is shown until the onload event of the iframe is called. Leaving this as is.